Tina cloud docs (#835)
tinacms/tinacms.org: content/guides/tina-cloud/existing-site/content-modelling.md

---
title: Model your content
last_edited: '2020-08-11T13:02:36.046Z'
---

> Now that we have tina set up on our site, we're almost able to start editing some content. Before we can do that, we will need to model our content within Tina Cloud.

Reviewer: ```suggestion
Now that we have Tina set up on our site, we're almost able to start editing some content. Before we can do that, we need to model our content within Tina Cloud.
```

The "Tina Graphql Gateway CLI" provides some tooling to help make this easy.

## Installation

The CLI can be installed as a dev dependency in your project.

```bash,copy
yarn add --dev tina-graphql-gateway-cli
```

## Getting started

The simplest way to get started is to add a `.tina/schema.ts` file

```bash,copy
mkdir .tina && touch .tina/schema.ts
```

This schema will need to include each of the fields that are in our \_posts files.

> In .tina/schema.ts, copy/paste the following:

Reviewer: ```suggestion
In `.tina/schema.ts`, copy/paste the following:
```

              label: 'Date',
              name: 'date',
            },
            {
              type: 'group',
              label: 'Author',
              name: 'author',
              fields: [
                {
                  type: 'text',
                  label: 'Name',
                  name: 'name',
                },
                {
                  type: 'text',
                  label: 'Picture',
                  name: 'picture',
                },
              ],
            },
            {
              type: 'group',
              label: 'OG Image',
              name: 'ogImage',
              fields: [
                {
                  type: 'text',
                  label: 'Url',
                  name: 'url',
                },
              ],
            },
          ],
        },
      ],
    },
  ],
})
```

You will also need to add this field/value to each of markdown file in "\_posts":

`_template: "post"`

Now that we have defined a "post" model, and applied it to each of our posts.

The last step to setting up our content models is to "compile" our schema. Through the terminal, run:

```bash,copy
yarn tina-gql schema:compile
```

> Once that is successful, make sure to **push everything up to your GitHub repo** (This is an important step, as the Tina Cloud content api will read this schema from your repository).

Reviewer: ```suggestion
Once that is successful, **push your changes up to your GitHub repository** 
This is an mandatory step, as the Tina Cloud content API reads this schema from your GitHub repository.
```
